### PR TITLE
Generate ctor for target typed new

### LINKED ISSUE #47928: GenerateConstructor doesn't trigger for target typed new
Breaking #42559 into two issues for clarity.

From the original issue:

> GenerateConstructor doesn't trigger on C c = new(1, 2, 3, 4); to offer to add a constructor with 4 parameters.

### PR DESCRIPTION
**Pull request by Youssef1313**
Currently this will crash. Will need to find a way to deal with delegated constructors.
The current implementation depends on retrieving the TypeSyntax and doing some logic on it. The implicit object creation doesn't have a TypeSyntax, so the implementation will need some updates to use semantic model.

I'll try to put more time on this to get it fixed.

Fixes #47928

### CODE REVIEW

**Comment by Youssef1313** [comment]
@allisonchou Since the issue is assigned to you, I'm okay if you got the time and want to take it over and continue fixing the delegating ctor part.

**Comment by allisonchou** [comment]
@Youssef1313 Don't worry about the assignment - if you want to continue working on the issue, that's totally OK with me. I haven't had the time to really dig into this one yet anyway. 😊

**Comment by Youssef1313** [comment]
@allisonchou I couldn't get my head around the existing logic for delegating constructors. So just wanted to note that you can take it over if you got the time to (i.e. I don't want to turn the issue stale due to a PR that may not move forward). In the meanwhile, I'll give it a second try.

File: src/Features/CSharp/Portable/GenerateConstructor/CSharpGenerateConstructorService.cs [hunk 1]
```diff
@@ -41,6 +41,9 @@ protected override bool IsSimpleNameGeneration(SemanticDocument document, Syntax
         protected override bool IsConstructorInitializerGeneration(SemanticDocument document, SyntaxNode node, CancellationToken cancellationToken)
             => node is ConstructorInitializerSyntax;
 
+        protected override bool IsTargetTypedNewGeneration(SemanticDocument document, SyntaxNode node, CancellationToken cancellationToken)
```
Comment (alrz) [inline src/Features/CSharp/Portable/GenerateConstructor/CSharpGenerateConstructorService.cs:44]: >TargetTypedNew

:memo: The standard term will be ImplicitObjectCreation from now on (PR 48395)

**Comment by davidwengier** [comment]
Put up https://github.com/dotnet/roslyn/pull/48814 with support for delegating constructors if you're curious. It was painful to understand, so don't feel bad about not getting it :)

**Comment by Youssef1313** [comment]
Thanks for taking it over. Closing in favor of #48814

**Pull request closed without merging**